src/web/tools/PointerEventManager.ts

```diff
@@ -0,0 +1,125 @@
+import { AdaptedEvent, EventTypes, MouseButtons } from '../interfaces';
+import EventManager from './EventManager';
+
+export default class PointerEventManager extends EventManager {
+  public setListeners(): void {
+    this.view.addEventListener('pointerdown', (event: PointerEvent): void => {
+      if (event.pointerType === 'touch') {
+        return;
+      }
+
+      if (!this.isPointerInBounds({ x: event.clientX, y: event.clientY })) {
+        return;
+      }
+
+      const adaptedEvent: AdaptedEvent = this.mapEvent(event, EventTypes.DOWN);
+      const target = event.target as HTMLElement;
+
+      target.setPointerCapture(adaptedEvent.pointerId);
+      this.markAsInBounds(adaptedEvent.pointerId);
+
+      if (++this.activePointersCounter > 1) {
+        adaptedEvent.eventType = EventTypes.ADDITIONAL_POINTER_DOWN;
+        this.onPointerAdd(adaptedEvent);
+        return;
+      }
+
+      this.onPointerDown(adaptedEvent);
+    });
+
+    this.view.addEventListener('pointerup', (event: PointerEvent): void => {
+      if (event.pointerType === 'touch') {
+        return;
+      }
+
+      if (this.activePointersCounter === 0) {
+        return;
+      }
+
+      const adaptedEvent: AdaptedEvent = this.mapEvent(event, EventTypes.UP);
+      const target = event.target as HTMLElement;
+
+      target.releasePointerCapture(adaptedEvent.pointerId);
+      this.markAsOutOfBounds(adaptedEvent.pointerId);
+
+      if (--this.activePointersCounter > 0) {
+        adaptedEvent.eventType = EventTypes.ADDITIONAL_POINTER_UP;
+        this.onPointerRemove(adaptedEvent);
+        return;
+      }
+
+      this.onPointerUp(adaptedEvent);
```

Comment:
```suggestion
      if (--this.activePointersCounter > 0) {
        adaptedEvent.eventType = EventTypes.ADDITIONAL_POINTER_UP;
        this.onPointerRemove(adaptedEvent);
      } else {
        this.onPointerUp(adaptedEvent);
      }
```

This may be more of a personal preference, but this looks more obvious to me. We can talk about it if you have a different opinion.

Reply:
Could be, but in this case I agree, that the ones You suggest look better